### PR TITLE
Pancake renames

### PR DESCRIPTION
**Pull request by jakozaur**
As part of the pancake cleanup, I implement @pdelewski and @nablaone comments. Still any suggestions including @trzysiek highly welcomed.

We want to split the code into phases:
1. Generate a pancake aggregation tree (eventually, we can even rename it to aggregation tree).
2. Transform aggregation query tree into pancake layered model.
3. Generate SQL query out of pancakes.

Still to be done in follow-up queries:
- [ ] cleanup logic to generate SQLs

### CODE REVIEW

File: quesma/queryparser/pancake_aggregation_tree.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import "quesma/model"
+
+// Structs used to represent the aggregation tree generated by first phase of pancake.
+type pancakeAggregationTree struct {
```
Comment (nablaone) [inline quesma/queryparser/pancake_aggregation_tree.go:8]: The list of trees is called a forest ;)

**Review by nablaone — APPROVED** [review]
LGTM.

File: quesma/queryparser/pancake_model.go [hunk 2]
```diff
@@ -37,7 +30,7 @@ type pancakeFillingMetricAggregation struct {
 	metadata model.JsonMap
 }
 
-type pancakeLayerBucketAggregation struct {
+type pancakeModelBucketAggregation struct {
 	name            string          // as originally appeared in Query DSL
```
Comment (pdelewski) [inline quesma/queryparser/pancake_model.go:34]: Some fields are common between metric and bucket aggregations. Maybe we could merge them? or embed one type in another

File: quesma/queryparser/pancake_aggregation_parser.go [hunk 3]
```diff
@@ -14,12 +14,16 @@ import (
 
 const PancakeOptimizerName = "pancake"
 
-// Here is experimental code to generate aggregations in one SQL query. called Version Una.
+// New way of generating queries, based on pancake model:
```
Comment (pdelewski) [inline quesma/queryparser/pancake_aggregation_parser.go:17]: This kind of comments helps a lot

**Review by pdelewski — APPROVED** [review]

File: quesma/queryparser/pancake_model.go (hunk 2 above)
Comment (jakozaur) [reply quesma/queryparser/pancake_model.go:34]: I agree, but let me do that in the following PR.

**Pull request merged**